sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -332,6 +332,9 @@ private[sql] abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
     def apply(plan: LogicalPlan): Seq[SparkPlan] = plan match {
       case r: RunnableCommand => ExecutedCommand(r) :: Nil
 
+      case _: logical.ScriptTransformation =>
```

Comment:
This is a quite far down the line for catching such an problem.

We can also throw an exception in the `CatalystSqlParser`s 'withScriptIOSchema' function. We would have to move the 'transform query spec' tests in the PlanParserSuite to Hive (HiveQlSuite).